src/capi/azext_capi/custom.py

```diff
@@ -226,7 +226,7 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
         control_plane_machine_count=os.environ.get("AZURE_CONTROL_PLANE_MACHINE_COUNT", 3),
         node_machine_type=os.environ.get("AZURE_NODE_MACHINE_TYPE"),
         node_machine_count=os.environ.get("AZURE_NODE_MACHINE_COUNT", 3),
-        kubernetes_version=os.environ.get("AZURE_KUBERNETES_VERSION", "1.20.6"),
+        kubernetes_version=os.environ.get("AZURE_KUBERNETES_VERSION", "1.22.8"),
```

Comment:
were you able to make sure that this works? I realized after we talked that 1.22 might not be supported with the old templates (it should according to https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions but good to double check)